libsolidity/AST.h

```diff
@@ -385,10 +447,15 @@ class VariableDeclaration: public Declaration
 	void setType(std::shared_ptr<Type const> const& _type) { m_type = _type; }
 
 	virtual LValueType getLValueType() const override;
+	bool isLocalVariable() const { return !!dynamic_cast<FunctionDefinition const*>(getScope()); }
+	bool isPublic() const { return m_isPublic; }
+	bool isStateVariable() const { return m_isStateVariable; }
 
-private:
-	ASTPointer<TypeName> m_typeName; ///< can be empty ("var")
 
+private:
+	ASTPointer<TypeName> m_typeName;    ///< can be empty ("var")
+	bool m_isPublic;                    ///< Whether there is an accessor for it or not
+	bool m_isStateVariable;             ///< Whether or not this is a contract state variable
 	std::shared_ptr<Type const> m_type; ///< derived type, initially empty
```

Comment:
Please preserve the space above `m_type`. The (admittedly unwritten) convention is that everything above the first space is known at construction time (i.e. deducible from the local context) and attributes below it are set later, e.g. during the type deduction phase.